app/clients/cbc_proxy.py

```diff
@@ -100,27 +108,28 @@ def cancel_broadcast(
     ):
         pass
 
-    def _invoke_lambda(self, payload):
-        if not self.lambda_name:
-            current_app.logger.warning(
-                '{self.__class__.__name__} tried to send {payload} but cbc proxy aws env vars not set'
-            )
-            return
-
+    def _invoke_lambda_with_failover(self, payload):
         payload_bytes = bytes(json.dumps(payload), encoding='utf8')
+        result = self._invoke_lambda(self.lambda_name, payload_bytes)
+
+        if 'FunctionError' in result:
+            if result['Payload']['errorType'] == "CBCNewConnectionError":
+                result = self._invoke_lambda(self.failover_lambda_name, payload_bytes)
+            else:
+                raise CBCProxyException('Function exited with unhandled exception')
 
+        return result
+
+    def _invoke_lambda(self, lambda_name, payload_bytes):
         result = self._lambda_client.invoke(
-            FunctionName=self.lambda_name,
+            FunctionName=lambda_name,
             InvocationType='RequestResponse',
             Payload=payload_bytes,
         )
 
         if result['StatusCode'] > 299:
```

Comment:
Should we take the same approach for a 299 status code and try with the second lambda and if that fails then we raise the exception?

Reply:
yeah, assuming i'm reading the docs right and correlating this `StatusCode` with these potential errors: https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html#API_Invoke_Errors

if we accidentally misconfigure one of the lambdas, we should retry the other one.